File: SchoolWebApp-API/Project.Core/Entities/Class/SchoolClass.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.School;
using SchoolWebApp.Core.Entities.Shared;
using SchoolWebApp.Core.Entities.Staff;
using SchoolWebApp.Core.Entities.Students;

namespace SchoolWebApp.Core.Entities.Class
{
    public class SchoolClass : Base
    {
        public required string Name { get; set; }
        public string? Description { get; set; }

        public int LearningLevelId { get; set; }
        public LearningLevel LearningLevel { get; set; }

        public int SchoolStreamId { get; set; }
        public SchoolStream SchoolStream { get; set; }

        public int AcademicYearId { get; set; }
        public AcademicYear AcademicYear { get; set; }

        public List<SchoolClassLeaders> SchoolClassLeaders { get; set; }
        public List<Exam> Exams { get; set; }
        public List<StudentClass> StudentClasses { get; set; }
        public List<StaffSubject> StaffSubjects { get; set; }
    }
}

File: SchoolWebApp-Client/src/app/class/models/school-class.ts
import {ResourceModel} from '@/core/models/ResourceModel';
import {SchoolStream} from './school-stream';
import {LearningLevel} from '@/class/models/learning-level';
import {AcademicYear} from '@/school/models/academic-year';
import {Person} from '@/school/models/person';

export class SchoolClass extends ResourceModel<SchoolClass> {
    public name?: string;
    public rank?: string;
    public description?: string;

    public learningLevelId?: string;
    public learningLevel?: LearningLevel;

    public schoolStreamId?: string;
    public schoolStream?: SchoolStream;

    public academicYearId?: string;
    public academicYear?: AcademicYear;

    public schoolClassLeaders?: Person[];

    public isSelected?: boolean = false;

    constructor(model?: Partial<SchoolClass>) {
        super(model);
    }
}
